qiskit/wrapper/__init__.py

```diff
@@ -0,0 +1,151 @@
+from qiskit.backends.ibmq.ibmqprovider import IBMQProvider
+from qiskit.wrapper.defaultqiskitprovider import DefaultQISKitProvider
+
+# Default provider used by the rest of the functions on this module. Please
+# note that this is a global object.
+_DEFAULT_PROVIDER = DefaultQISKitProvider()
+
+
+def register(token, url,
+             hub=None, group=None, project=None, proxies=None, verify=True,
+             provider_name='qiskit'):
+    """
+    Authenticate against an online backend provider.
+
+    Args:
+            token (str): The token used to register on the online backend such
+                as the quantum experience.
+            url (str): The url used for online backend such as the quantum
+                experience.
+            hub (str): The hub used for online backend.
+            group (str): The group used for online backend.
+            project (str): The project used for online backend.
+            proxies (dict): Proxy configuration for the API, as a dict with
+                'urls' and credential keys.
+            verify (bool): If False, ignores SSL certificates errors.
+            provider_name (str): the unique name for the online backend
+                provider (for example, 'qiskit' for the IBM Quantum Experience).
+    Raises:
+        QISKitError: if the provider name is not recognized.
+    """
+    if provider_name == 'qiskit':
+        provider = IBMQProvider(token, url,
+                                hub, group, project, proxies, verify)
+        _DEFAULT_PROVIDER.add_provider(provider)
+    else:
+        raise QISKitError('provider name %s is not recognized' % provider_name)
+
+# Functions for inspecting and retrieving backends.
+
+
+def available_backends(filters=None):
+    """
+    Return the backends that are available in the SDK, optionally filtering
+    them based on their capabilities.
+
+    Note:
+        In order for this function to return online backends, a connection with
+        an online backend provider needs to be established by calling the
+        `register()` function.
+    Args:
+        filters (dict): dictionary of filtering conditions.
+    Returns:
+        list[str]: the names of the available backends.
+    """
+    return _DEFAULT_PROVIDER.available_backends(filters)
+
+
+def local_backends():
+    """
+    Return the available local backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': True})
+
+
+def remote_backends():
+    """
+    Return the available remote backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': False})
+
+
+def get_backend(name):
+    """
+    Return an instance of a `Backend` object from its identifier.
+
+    Args:
+        name(str): unique name of the backend.
+    Returns:
+        BaseBackend: a Backend instance.
+    """
+    return _DEFAULT_PROVIDER.get_backend(name)
```

Comment:
Similarly, this should query every provider for the named backend. It might have to return a list in some cases.